### PR TITLE
Password encryption and decryption

### LINKED ISSUE #603: Encryption and decryption supports for hotpot.
**Is this a BUG REPORT or FEATURE REQUEST?**:
> /kind feature

**What happened**:
Currently hotpot lacks encryption and decryption.

**What you expected to happen**:
The design ideas are as follows:
1.Add the following contents to the opensds.conf. This can be used to select the encryption and decryption kit. Currently only supports the aes encryption tool that has been implemented. Users can also implement it themselves.
```
# Encryption and decryption tool. Default value is aes.
password_decrypt_tool = aes
```
2.Provide two encryption and decryption tools. One is used for shell script calls to encrypt the password automatically, the other which does not echo the password is used for user to encrypt password that will be add to the configuration file.

3.All the passwords in the configuration files must be encrypted by encryption tool by default, otherwise the service will not start normally.

**Anything else we need to know?**:
Design proposal needs to be discussed as soon as possible.

**Environment**:
- Hotpot(release/branch) version: latest version

### PR DESCRIPTION
**Pull request by jackhaibo**
**What this PR does / why we need it**:

This PR provides a password security mechanism for OpenSDS. Please see [design spec](https://github.com/opensds/design-specs/pull/19) for more details.

**Which issue this PR fixes**: fixes #603 

**Special notes for your reviewer**:

@xing-yang @wisererik @leonwanghui Please take a review at this PR, thanks!

**Release note**: None

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,6 +42,9 @@ osdslet: prebuild
 osdsctl: prebuild
 	go build -o $(BUILD_DIR)/bin/osdsctl github.com/opensds/opensds/osdsctl
 
+pwdEncrypter: prebuild
```
Comment (leonwanghui) [inline Makefile]: This module should be designed as the same with `test` and `protoc` module.

Comment (jackhaibo) [reply Makefile]: Already updated, please review again.

File: pkg/utils/config/config_define.go [hunk 2]
```diff
@@ -19,15 +19,16 @@ import "time"
...
-	AuthStrategy        string        `conf:"auth_strategy,noauth"`
-	PolicyPath          string        `conf:"policy_path,/etc/opensds/policy.json"`
-	Daemon              bool          `conf:"daemon,false"`
-	LogFlushFrequency   time.Duration `conf:"log_flush_frequency,5s"` // Default value is 5s
-	HTTPSEnabled        bool          `conf:"https_enabled,false"`
-	BeegoHTTPSCertFile  string        `conf:"beego_https_cert_file,/opt/opensds-security/opensds/opensds-cert.pem"`
-	BeegoHTTPSKeyFile   string        `conf:"beego_https_key_file,/opt/opensds-security/opensds/opensds-key.pem"`
-	PasswordDecryptTool string        `conf:"password_decrypt_tool,aes"`
+	ApiEndpoint        string        `conf:"api_endpoint,localhost:50040"`
+	Graceful           bool          `conf:"graceful,true"`
```
Comment (leonwanghui) [inline pkg/utils/config/config_define.go]: Please remove `Graceful` and `SocketOrder` these two fileds.

Comment (jackhaibo) [reply pkg/utils/config/config_define.go]: Ask @xxwjj

Comment (jackhaibo) [reply pkg/utils/config/config_define.go]: Has been deleted.

**Pull request merged**